### PR TITLE
feat(flags): Add list and delete endpoints to signing-secrets resource

### PR DESCRIPTION
**Pull request by cmanallen**

### CODE REVIEW

File: src/sentry/flags/docs/api.md [hunk 1]
```diff
@@ -83,7 +83,26 @@ Retrieve a single flag log instance.
...
+Browse a list of signing secrets. Secrets are unique per provider. Secrets only show the first six characters; the remainder are redacted.
+
+- Response 200
+
+  ```json
+  {
+    "data": [
+      {
+        "createdAt": "2024-12-12T00:00:00Z",
+        "createdBy": 12345,
```
Comment (michellewzhang) [inline src/sentry/flags/docs/api.md:99]: is this the user ID?

File: src/sentry/flags/docs/api.md [hunk 2]
```diff
@@ -83,7 +83,26 @@ Retrieve a single flag log instance.
...
+        "provider": "launchdarkly",
+        "secret": "abc123**********"
+      }
+    ]
+  }
+  ```
 
 ### Create Signing Secret [POST]
```
Comment (michellewzhang) [inline src/sentry/flags/docs/api.md:107]: is the post endpoint different now? i see `/<provider>/` was removed

File: src/sentry/flags/docs/api.md [hunk 3]
```diff
@@ -99,6 +118,14 @@ Requests from web hook providers can be signed. We use the signing secret to ver
 
 - Response 201
 
+## Signing Secret [/organizations/<organization_id_or_slug>/flags/hooks/signing-secrets/<signing_secret_id>/]
```
Comment (michellewzhang) [inline src/sentry/flags/docs/api.md:121]: how do we get the signing secret id?

File: src/sentry/flags/docs/api.md (hunk 1 above)
Comment (cmanallen) [reply src/sentry/flags/docs/api.md:99]: Sentry user id yes.  Different name preferred?

File: src/sentry/flags/docs/api.md (hunk 2 above)
Comment (cmanallen) [reply src/sentry/flags/docs/api.md:107]: Yes! I neglected to add provider as a field in the payload.

File: src/sentry/flags/docs/api.md (hunk 3 above)
Comment (cmanallen) [reply src/sentry/flags/docs/api.md:122]: I'll return the id in the list response.  Updating now.

File: src/sentry/flags/docs/api.md (hunk 2 above)
Comment (michellewzhang) [reply src/sentry/flags/docs/api.md:108]: ok cool, i'll need to update this in the sidebar then👍

File: src/sentry/flags/docs/api.md (hunk 1 above)
Comment (michellewzhang) [reply src/sentry/flags/docs/api.md:99]: @ryan953 is there a way we can get a user name from the id on the frontend?

**Review by michellewzhang — APPROVED** [review]
looks good!

File: src/sentry/flags/docs/api.md (hunk 1 above)
Comment (ryan953) [reply src/sentry/flags/docs/api.md:116]: we could look up the user name on the server before returning the data, i think that's preferred to another round trip to the server.


but it looks like there is a `/users/$id` route as well that is implemented by `class UserDetailsEndpoint`

File: src/sentry/flags/endpoints/secrets.py [hunk 4]
```diff
@@ -8,23 +11,69 @@
...
+            "createdBy": obj.created_by,
+            "id": obj.id,
+            "provider": obj.provider,
+            "secret": secret,
+        }
+
+
 class FlagWebhookSigningSecretValidator(serializers.Serializer):
-    secret = serializers.CharField(required=True)
+    provider = serializers.ChoiceField(choices=[("launchdarkly", "launchdarkly")], required=True)
```
Comment (aliu39) [inline src/sentry/flags/endpoints/secrets.py:48]: dup launchdarkly

File: src/sentry/flags/endpoints/secrets.py [hunk 5]
```diff
@@ -8,23 +11,69 @@
...
+
+
+@register(FlagWebHookSigningSecretModel)
+class FlagWebhookSigningSecretSerializer(Serializer):
+    def serialize(self, obj, attrs, user, **kwargs) -> FlagWebhookSigningSecretResponse:
+        secret = obj.secret
+        if len(secret) >= 10:
+            secret = secret[0:6] + "*" * len(secret[6:])
+        else:
+            secret = "*" * len(secret)
```
Comment (aliu39) [inline src/sentry/flags/endpoints/secrets.py:36]: This conflicts with api.md:
> Secrets only show the first six characters; the remainder are redacted.

Sorry if I missed this, but do we have any kind of expectations/requirements on the length of secrets?

File: tests/sentry/flags/endpoints/test_secrets.py [hunk 6]
```diff
@@ -4,39 +4,131 @@
...
+            provider="launchdarkly",
+            secret="123456123456",
+        )
+        self.url = reverse(self.endpoint, args=(self.organization.id,))
 
     @property
     def features(self):
         return {"organizations:feature-flag-audit-log": True}
 
+    def test_browse(self):
```
Comment (aliu39) [inline tests/sentry/flags/endpoints/test_secrets.py:25]: Can we test the length < 10 case?

File: src/sentry/flags/endpoints/secrets.py (hunk 4 above)
Comment (cmanallen) [reply src/sentry/flags/endpoints/secrets.py:48]: Not a dupe just a bad API :\. Has to be this way...

File: tests/sentry/flags/endpoints/test_secrets.py (hunk 6 above)
Comment (cmanallen) [reply tests/sentry/flags/endpoints/test_secrets.py:25]: Remove that case.

File: src/sentry/flags/endpoints/secrets.py (hunk 5 above)
Comment (cmanallen) [reply src/sentry/flags/endpoints/secrets.py]: I just removed it 👍.  Behavior matches blueprint.

File: src/sentry/flags/endpoints/secrets.py [hunk 7]
```diff
@@ -29,24 +29,18 @@ class FlagWebhookSigningSecretResponse(TypedDict):
...
-        else:
-            secret = "*" * len(secret)
-
         return {
             "createdAt": obj.date_added.isoformat(),
             "createdBy": obj.created_by,
             "id": obj.id,
             "provider": obj.provider,
-            "secret": secret,
+            "secret": obj.secret[0:6] + "*" * (len(obj.secret) - 6),
```
Comment (aliu39) [inline src/sentry/flags/endpoints/secrets.py:37]: I think this will error if len < 6, multiplying string by negative number

Comment (aliu39) [reply src/sentry/flags/endpoints/secrets.py:37]: Ok nvm, python continues to impress 😂

File: src/sentry/flags/endpoints/secrets.py [hunk 8]
```diff
@@ -29,24 +29,18 @@ class FlagWebhookSigningSecretResponse(TypedDict):
...
         }
 
 
 class FlagWebhookSigningSecretValidator(serializers.Serializer):
     provider = serializers.ChoiceField(choices=[("launchdarkly", "launchdarkly")], required=True)
-    secret = serializers.CharField(required=True, max_length=128)
+    secret = serializers.CharField(required=True, max_length=32, min_length=32)
```
Comment (aliu39) [inline src/sentry/flags/endpoints/secrets.py:43]: nice, ignore my comment above about negative numbers 👍

**Review by aliu39 — APPROVED** [review]

File: src/sentry/flags/endpoints/secrets.py (hunk 7 above)
Comment (cmanallen) [reply src/sentry/flags/endpoints/secrets.py:37]: Its validated to be length 32.

Comment (cmanallen) [reply src/sentry/flags/endpoints/secrets.py:37]: Oh nvm just saw your other comment :D

**Pull request merged**